src/ResourceManager/Compute/Commands.Compute/Models/AzureDiskEncryptionStatusContext.cs

```diff
@@ -3,16 +3,24 @@
 
 namespace Microsoft.Azure.Commands.Compute.Models
 {
+    enum EncryptionStatus
+    {
+        Encrypted,
+        NotEncrypted,
+        Unknown
+    }
+
     class AzureDiskEncryptionStatusContext
     {
-        public bool OsVolumeEncrypted { get; set; }
+        public EncryptionStatus OsVolumeEncrypted { get; set; }
         public DiskEncryptionSettings OsVolumeEncryptionSettings { get; set; }
-
-        [JsonIgnore]
-        public string OsVolumeEncryptionSettingsText
-        {
-            get { return JsonConvert.SerializeObject(OsVolumeEncryptionSettings, Formatting.Indented); }
-        }
-        public bool DataVolumesEncrypted { get; set; }
+        public EncryptionStatus DataVolumesEncrypted { get; set; }
+    }
+    class AzureDiskEncryptionStatusLinuxContext
+    {
```

Comment:
On Windows the DataVolumeEncryptionSettings are not -- logically speaking -- "null". They are derived/same as the OsVolumeEncryptionSettings. It will be confusing for users when they inspect the object and see null for DataVolumeEncryptionSettings. Instead, on Windows, it makes sense to not have a DataVolumeEncryptionSettings at all.